helix-core/src/match_brackets.rs

```diff
@@ -16,40 +16,55 @@ pub fn find(syntax: &Syntax, doc: &Rope, pos: usize) -> Option<usize> {
 
     let byte_pos = doc.char_to_byte(pos);
 
-    // most naive implementation: find the innermost syntax node, if we're at the edge of a node,
-    // return the other edge.
+    // a little less naive implementation: find the innermost syntax node at
+    // the given byte position. Traverse up the syntax tree until we reach the
+    // first node with surrounding pairs.
```

Comment:
You'll probably want to keep both versions of the function, because it's used to find and highlight the matching bracket when going over a bracket: https://github.com/helix-editor/helix/blob/90fd09f2cc4e5dfa7380036b0748adf95ffdf371/helix-term/src/ui/editor.rs#L380

Reply:
Good point. The highlighting sill works, but it also highlights the closing bracket of the current block your cursor is in. This is probably not ideal. I'll change it.